File: src/main/java/com/itmo/microservices/shop/user/api/service/UserService.java
package com.itmo.microservices.shop.user.api.service;

import com.itmo.microservices.shop.user.api.model.RegistrationRequest;
import com.itmo.microservices.shop.user.api.model.UserModel;

import java.util.UUID;

public interface UserService {
    UserModel registerUser(RegistrationRequest request);

    UserModel getUserByID(UUID uuid);

    UserModel getUserByUsername(String username);
}

File: src/main/kotlin/com/itmo/microservices/shop/demo/users/api/service/UserService.kt
package com.itmo.microservices.shop.demo.users.api.service

import com.itmo.microservices.shop.demo.users.api.model.AppUserModel
import com.itmo.microservices.shop.demo.users.api.model.RegistrationRequest
import org.springframework.security.core.userdetails.UserDetails

interface UserService {
    fun findUser(username: String): AppUserModel?
    fun registerUser(request: RegistrationRequest)
    fun getAccountData(requester: UserDetails): AppUserModel
    fun deleteUser(user: UserDetails)
}